### ISSUE TITLE
No support for *time.Time

### ISSUE THREAD

**Issue by bikbah** [go-playground/form#4]
There is no support for `*time.Time`, but there is for `time.Time`

**Comment by joeybloggs** [comment]
Hello @bikbah 

`*time.Time` should be supported, It is tested for within the tests [here](https://github.com/go-playground/form/blob/master/decoder_test.go#L507), [here](https://github.com/go-playground/form/blob/master/decoder_test.go#L534) and [here](https://github.com/go-playground/form/blob/master/decoder_test.go#L607)

perhaps you could provide a code sample where it is not working, it's possible I missed a scenario.

P.S.
It is also possible that it just looks like it's not supported:
eg. when calling [ExtractType](https://github.com/go-playground/form/blob/master/decoder.go#L191) it dereferences the pointer before the kind switch statement and so all code dealing with time will  have a dereferences `time.Time` instead of `*time.Time`

so in effect, when registering a custom type, you should only need to add the non-pointer type value.
eg.
```go
decoder.RegisterCustomTypeFunc(func(vals []string) (interface{}, error) {
		return time.Parse("2006-01-02", vals[0])
	}, time.Time{})
```

**Comment by bikbah** [comment]
I am really sorry. :( It works for pointers. I was passing empty string and trying to parse it as proper `time.Time`.

**Comment by joeybloggs** [comment]
No problems @bikbah, if you ever have any questions or think there's an issue don't hesitate to ask :)

**Comment by bikbah** [comment]
OK, I found bug. There is no initialization of pointer to `time.Time` field on default decoder even for valid time string. Here are my tests, put it to `decoder_test.go` and run. `TestDecoderTimePointerDefault` fails, but `TestDecoderTimePointerCustom` does not.

`go
func TestDecoderTimePointerDefault(t *testing.T) {
	type TestStruct struct {
		Time    time.Time
		TimePtr *time.Time
	}

	values := url.Values{
		"Time":    []string{"2016-01-02T15:04:05Z"},
		"TimePtr": []string{"2016-01-02T15:04:05Z"},
	}

	var test TestStruct

	decoder := NewDecoder()

	errs := decoder.Decode(&test, values)
	Equal(t, errs, nil)

	tm, _ := time.Parse(time.RFC3339, "2016-01-02T15:04:05Z")
	Equal(t, test.Time.Equal(tm), true)
	NotEqual(t, test.TimePtr, nil)
	Equal(t, (*test.TimePtr).Equal(tm), true)
}

func TestDecoderTimePointerCustom(t *testing.T) {
	type TestStruct struct {
		Time    time.Time
		TimePtr *time.Time
	}

	values := url.Values{
		"Time":    []string{"2016-01-02T15:04:05Z"},
		"TimePtr": []string{"2016-01-02T15:04:05Z"},
	}

	var test TestStruct

	decoder := NewDecoder()
	decoder.RegisterCustomTypeFunc(func(vals []string) (interface{}, error) {
		if vals[0] == "" {
			return time.Time{}, nil
		}
		return time.Parse(time.RFC3339, vals[0])
	}, time.Time{})

	errs := decoder.Decode(&test, values)
	Equal(t, errs, nil)

	tm, _ := time.Parse(time.RFC3339, "2016-01-02T15:04:05Z")
	Equal(t, test.Time.Equal(tm), true)
	NotEqual(t, test.TimePtr, nil)
	Equal(t, (*test.TimePtr).Equal(tm), true)
}
`

**Comment by joeybloggs** [comment]
I'll take a look right away

**Comment by joeybloggs** [comment]
OK @bikbah 

thanks for reporting, I have corrected in commit 762ce96aa9e35c0f3e10fe0265e6106e59ff9a03, please pull the latest code.

fortunately it wasn't a logic problem, I merely forgot to set the `set` variable to true after assigning, so the proceeding code though there was nothing to set on the struct field. [see here](https://github.com/go-playground/form/blob/master/decoder.go#L557) for the fix, it's literally this line was missing.

Thanks for reporting 👍

**Comment by bikbah** [comment]
Yeah, it works well. Thanks for quick response. :)
